README.md

```diff
@@ -1,37 +1,107 @@
 [![Coverty Scan](https://scan.coverity.com/projects/13163/badge.svg)](https://scan.coverity.com/projects/stlab_libraries)
 
-# Content
+## Content
 
-## [Concurrency](https://www.stlab.cc/libraries/concurrency/)
+### [Concurrency](https://www.stlab.cc/libraries/concurrency/)
 This library provides futures and channels, high level abstractions for implementing algorithms that eases the use of multiple CPU cores while minimizing contention. This library solves several problems of the C++11 and C++17 TS futures.
 
-# Documentation
+## Documentation
 
 The complete documentation is available on the [STLab home page](http://stlab.cc).
 
 Release changelogs are listed in [CHANGES.md](CHANGES.md).
 
-# Supported Platforms
-- All OS that have a C++14 compliant compiler
+## Tested on
 
-# Requirements
-- C++14 Compiler
--- Visual Studio 2019
--- gcc (>= 7)
--- clang (>= 6)
-- [OPTIONAL] Boost.Optional and Boost.Variant. Required when `STLAB_USE_BOOST_CPP17_SHIMS` is enabled.
-- [OPTIONAL] Boost.Test. Used when `BUILD_TESTING` is enabled.
+- Linux with GCC 11
+- Linux with Clang 14
+- MacOS with Apple-clang 13.0.1
+- Windows with Visual Studio 16
 
-# Building
+## Requirements
 
-STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial for an introduction to this tool.
+- A standards-compliant C++14, C++17, or C++20 compiler
+- **Use with a compiler in C++14-compliant mode** requires Boost.Optional and Boost.Variant >= 1.74.0
+- **Building** the library requires CMake 3.23 or later
+- **Testing or developing** the library requires Boost.Test >= 1.74.0
+
+## Building
+
+STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial
+for an introduction to this tool.
+
+### Preparation
+
+1. Create a build directory outside this library's source tree.  In this guide, we'll use a sibling
+   directory called `BUILD`.
+
+2. If you are using the library in C++14-compliant mode or need to run the test suite, be sure you
+   have the necessary parts of Boost >= 1.74.0 installed.  Linux distributions usually make a
+   suitable version available through standard package managers.  On macOS or Linux, you can easilly
+   install Boost using [Homebrew](https://brew.sh/).  To install Boost on Windows, you can use
+   Boost's [binary installers](https://sourceforge.net/projects/boost/files/boost-binaries/).
+
+3. Install a version of CMake >= 3.23.  If you are on Debian or Ubuntu Linux you may need to use
+   `snap` to find one that's new enough.
+
+4. If you are using MSVC, you may need to set environment variables appropriately for your target
+   architecture by invoking `VCVARSALL.BAT` with an appropriate option.
+
+### Configure
+
+Run CMake in the root directory of this project, setting `../BUILD` as your build directory.  The
+basis of your command will be
+
+```
+cmake -S . -B ../BUILD
```

Comment:
I thought about this carefully. It is not so unconventional; I've used CMake for years both ways. It's what our GH actions do.  And as long as this repo is called `libraries`, the user is likely to check it out with a different name, and then these instructions get more complicated or error-prone because they have to refer to a directory whose name we don't know.

Reply:
```suggestion
cmake -S . -B ../BUILD # + other options. See below...
```

Reply:
Okay, I don't feel strongly about it.